### ISSUE TITLE
Error: Could not update file: 4 of 4 required status checks are expected.

### ISSUE THREAD

**Issue by EmilStenstrom** [akhilmhdh/contributors-readme-action#28]
Hi,

I'm getting a new kind of error. I think it's because I accepted a PR from dependabot. What does "Error: Could not update file: 4 of 4 required status checks are expected." mean, and do you have any idea of what I can do to resolve it?

Link to relevant error:
https://github.com/EmilStenstrom/django-components/runs/3475644213?check_suite_focus=true

**Comment by EmilStenstrom** [comment]
Ok, so this does not seem to be generated by this library directly: https://github.com/akhilmhdh/contributors-readme-action/search?q=console

**Comment by EmilStenstrom** [comment]
So this must have something to do with my repo having status checks defined. 

![image](https://user-images.githubusercontent.com/224130/132177300-361cfa56-2afc-423e-a282-55b3965a79d1.png)

**Comment by EmilStenstrom** [comment]
The only thing I can think of is that this action merges too fast, so status checks have not be run yet, and therefore fails. Any ideas of how to solve this?

Maybe this? https://github.com/WyriHaximus/github-action-wait-for-status

**Comment by akhilmhdh** [comment]
Ohhh damn. 6 days ago. Oopz really missed the notification. My bad first of all sorry for the delayed response.

**Comment by EmilStenstrom** [comment]
@akhilmhdh Sorry for rushing, wasn't my intent! I just wanted to add a bit more context and see if I could solve this myself.

**Comment by akhilmhdh** [comment]
Not at all. Thank you for looking through this while I was absent. Yah your right I think its due to the status check.

Didn't knew github even had a feature like this 😅 .
I think its may be i am directly committing and pushing the readme change.

**Comment by akhilmhdh** [comment]
can you try using with the above action. But I don't think it will work as I am using Github api to update the readme.

One way to solve this is to run the action in cron on another branch and then merge it periodically.

**Comment by EmilStenstrom** [comment]
I think status checks are available in the API too. So maybe there could be another API call to check if the check(s) has passed or not? https://docs.github.com/en/rest/reference/checks

I've seen lots of projects that require tests to pass to allow commits to the main branch, that's what I'm using this for. For now, I've disabled status checks so that I get proper contributor attribution in the readme.

**Comment by akhilmhdh** [comment]
@EmilStenstrom  good find. Give me this week. Let me see if this addressed in API and if so I ll make that change. In meantime if you find anything please update me. Would be happy to help you

**Comment by akhilmhdh** [comment]
@EmilStenstrom  I think I did found a solution. The issue with status check or any branch protection is direct commit is not allowed. So how status check works is, after a commit to another branch and on putting a PR the status check runs. Only after that commit can be made.

So i am planning to soon integrate a new feature in which for protected branches the action will put a PR for any change in contributors list. I think github api has support for this

**Comment by EmilStenstrom** [comment]
Nice, that's a great solution! Looking forward to it.

**Comment by EmilStenstrom** [comment]
@akhilmhdh No worries, take your time. I've disabled status checks for my repo while you are working on this.

**Comment by akhilmhdh** [comment]
@EmilStenstrom  This is the workflow that i came up with....

When there is change in contributors list these are the changes that comes up when protection mode is set true..

1. A branch is created with `contributors-action-<timestamp>`
2. Then the readme PR is made with a customizable title and description
3. You can then accept the PR. 

So for all new changes the PR it will create a new branch to avoid  conflict. To avoid so many branches you can either manual delete on PR merge or set the your repo setting to automatically delete branch on merge.

Is this workflow good, or do you have any suggestion

**Comment by EmilStenstrom** [comment]
@akhilmhdh Sounds great to me! Thanks for working on this! :)

**Comment by akhilmhdh** [comment]
Awesome. Will soon release this version

**Comment by akhilmhdh** [comment]
Hey @EmilStenstrom, I think this took longer than I thought. 
First of all thank you so much for your patience. 

I have released the the new version of [v2.3](https://github.com/marketplace/actions/contribute-list) which has the `is_protected` option that will now onwards send a PR on update in contributors list on setting it to true.

Closing this issue as this will solve it. Feel free to re-open it if it's not solving. 😄 

Sidenote: Added you to my action's [feature thank away list](https://github.com/akhilmhdh/contributors-readme-action/blob/master/contributors.md).
